src/vstest.console/CommandLine/Executor.cs

```diff
@@ -329,11 +329,19 @@ private bool ExecuteArgumentProcessor(IArgumentProcessor processor, ref int exit
             }
             catch (Exception ex)
             {
-                if (ex is CommandLineException || ex is TestPlatformException || ex is SettingsException)
+                if (ex is CommandLineException || ex is TestPlatformException || ex is SettingsException || ex is InvalidOperationException)
                 {
                     EqtTrace.Error("ExecuteArgumentProcessor: failed to execute argument process: {0}", ex);
                     this.Output.Error(false, ex.Message);
                     result = ArgumentProcessorResult.Fail;
+
+                    // Send inner exception only when its message is different to avoid duplicate.
+                    if (ex is TestPlatformException &&
```

Comment:
Instead of this we can do ex.ToString() in line 335 which will print inner exception. But in that case duplicacy will be there if inner exception and outer exception message are same. This happens for many scenarios in TestPlatformException